src/btree/bt_page.c

```diff
@@ -15,6 +15,434 @@ static int __inmem_row_int(WT_SESSION_IMPL *, WT_PAGE *, size_t *);
+    page = ref->page;
+    base_entries = (size_t)page->dsk->u.entries;
+
+    WT_ASSERT(session, base_entries != 0 && delta_entries != 0);
+
+    /* Unpack all entries from the base image into an array. */
+    WT_ERR(__wt_calloc_def(session, base_entries, &base));
+    WT_CELL_FOREACH_ADDR (session, page->dsk, base[k]) {
+        k++;
+    }
+    WT_CELL_FOREACH_END;
+
+    /*
+     * Creates a new reference array containing the finalized refs. The maximum number of entries is
+     * the sum of half the base entries (since keys and values are stored separately) and the delta
+     * entries.
+     */
+    estimated_entries = (base_entries / 2) + delta_entries + 1;
+    WT_ERR(__wt_calloc_def(session, estimated_entries, refsp));
+    refs = *refsp;
+
+    /* Perform a merge sort between the base array and the delta array. */
+    while (i < base_entries && j < delta_entries) {
+        /* Compare the keys of the base entry and the delta entry. */
+        base_key_buf.data = base[i].data;
+        base_key_buf.size = base[i].size;
+        delta_key_buf.data = delta[j]->key.data;
+        delta_key_buf.size = delta[j]->key.size;
+        WT_ERR(__wt_compare(session, S2BT(session)->collator, &base_key_buf, &delta_key_buf, &cmp));
+
+        if (cmp < 0) {
+            base_key = &base[i++];
+            base_val = &base[i++];
+            WT_ERR(__page_build_ref(
+              session, ref, base_key, base_val, NULL, true, &refs[final_entries++], incr));
+        } else if (cmp >= 0) {
+            if (!F_ISSET(delta[j], WT_DELTA_INT_IS_DELETE))
+                WT_ERR(__page_build_ref(
+                  session, ref, NULL, NULL, delta[j], false, &refs[final_entries++], incr));
+            if (cmp == 0)
+                i += 2; /* Skip the current key and value. */
+            j++;
+        }
+    }
+    /* Copy the remaining entries from the base array or the delta array. */
+    for (; i < base_entries;) {
+        base_key = &base[i++];
+        base_val = &base[i++];
+        WT_ERR(__page_build_ref(
+          session, ref, base_key, base_val, NULL, true, &refs[final_entries++], incr));
+    }
+    for (; j < delta_entries; j++)
+        if (!F_ISSET(delta[j], WT_DELTA_INT_IS_DELETE))
+            WT_ERR(__page_build_ref(
+              session, ref, NULL, NULL, delta[j], false, &refs[final_entries++], incr));
+
+    WT_ASSERT(session, i == base_entries && j == delta_entries);
+    WT_ASSERT(session, final_entries != 0);
+    WT_ASSERT(session, final_entries < estimated_entries && refs[final_entries] == NULL);
+
+err:
+    *ref_entriesp = final_entries;
+    __wt_free(session, base);
+    return (ret);
+}
+
+/*
+ * __page_unpacked_delta_key_cmp --
+ *     Compare two unpacked deltas
+ */
+static int
+__page_unpacked_delta_key_cmp(
+  WT_SESSION_IMPL *session, const WT_CELL_UNPACK_DELTA_INT *a, const WT_CELL_UNPACK_DELTA_INT *b)
+{
+    WT_DECL_RET;
+    WT_ITEM key_a, key_b;
+    int cmp;
+
+    key_a.data = a->key.data;
```

Comment:
Here I still don't understand why key_a key_b is needed.

Can we just do:

__wt_compare(session, S2BT(session)->collator, a->key, b->key, &cmp))?

Reply:
__wt_compare expects a `WT_ITEM`, but a->key is `WT_CELL_UNPACK_KV` and val is `WT_CELL_UNPACK_ADDR`